identity-wallet/src/state/mod.rs

```diff
@@ -46,7 +46,7 @@ pub struct AppState {
     pub locale: Locale,
     pub credentials: Vec<DisplayCredential>,
     pub current_user_prompt: Option<CurrentUserPrompt>,
-    pub dev_mode_enabled: bool,
+    pub dev_profile: Option<ProfileType>,
```

Comment:
Through introducing this optional enum, you will now have states such as:

```json
{
  "active_profile": {
    "name": "Shenron"
  },
  "dev_profile": "Shenron"
}
```

I assume from the logic of the rest of the code, that this state would never be produced for example (since it's inconsistent):

```json
{
  "active_profile": {
    "name": "John Doe"
  },
  "dev_profile": "Ferris"
}
```

Why have a `ProfileType` that distinguishes between more types than just "there is a dev profile loaded" / "there is no dev profile loaded"? This would make it a boolean and we could just call it `dev_profile_active` or something and automatically inject the dev profile password based on that boolean value.

Reply:
Another option: we could also just add another field to `active_profile` such as `dev_profile: bool` and default to `false`. This is where it would make the most sense. You would read the profile anywhere in the code and then make distinctions based on that inner flag of the profile itself. This would hurt the business requirement to "keep the dev mode out of other structures as much as possible" though.

Reply:
I will make a different enum on the state to differentiate between auto login and not. So 
DevMode {
  On,
  Off,
  OnWithAutoLogin
}

Reply:
So I guess the reason why `dev_mode_enabled` 'cannot' be a `bool`, is because we need to make the distinction between 1. a regular profile enabling `dev_mode` and a 2. Dev Profile (that per definition is in `dev_mode`).
1. When opening the app, the `get_state` reducer should set: `state.current_user_prompt = Some(CurrentUserPrompt::PasswordRequired);` so that the user is forced to enter their password to unlock stronghold, etc..
2. For Dev Profiles, there should not be a CurrentUserPrompt, instead the storage should be automatically unlocked using the `PASSWORD` const. 

To add a bit more context: `fn get_state` just loads a state, including the `active_profile`. It is totally unaware if this `active_profile` is a  regular profile in `dev_mode` or a Dev Profile in `dev_mode`. 

I think that's the core of it, the problem to be solved is: how do we distinguish between the two? (please correct me if I'm wrong).

I think @Norlock suggestion is a viable option.

Reply:
I have implemented the suggestion above